### PR TITLE
CCM-1704: Address sonar issues

### PR DESCRIPTION
**Pull request by simonlabarere**
## Summary
Go through the list of recommended updates by Sonar https://sonarcloud.io/summary/overall?id=NHSDigital_communications-manager-api
Note that Sonar uses `master` as the main branch so this changes will only be reflected once this is released.

Ignores:
```"links": Object.assign({}, { "about": "https://digital.nhs.uk/developer/api-catalogue/communications-manager" }, links),```
I couldn't use the following as recommended as it's not compatible with Apigee
```"links": {"about": "https://digital.nhs.uk/developer/api-catalogue/communications-manager", ...links},```

We decided to keep fixtures as start imports as this is not production code and it's only used to run end to end tests in a pipeline. It saves a lot time and it avoids having to deal with dependencies.
```from lib.fixtures import *```

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [x] Brief description of work completed, and any technical decisions made as part of the PR
* [x] PR link added as a comment to the relevant JIRA ticket
* [x] PR link shared on Slack and/or Teams
* [x] 2 reviews received
* [x] Tester approval

### CODE REVIEW

File: sandbox/handlers/batch_send.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ export async function batch_send(req, res, next) {
 
   // Note: the docker container uses node:12 which does not support optional chaining
   const odsCodes = messages.map((message) => {
-    if (message && message.originator) {
+    if (message?.originator) {
```
Comment (aaron-downing) [inline sandbox/handlers/batch_send.js:88]: I think this would suffice
```suggestion
   return message?.originator?.odsCode
```

Comment (aaron-downing) [reply sandbox/handlers/batch_send.js:88]:
```
const odsCodes = messages.map((message) => ( message?.originator?.odsCode ))
```

File: sandbox/handlers/utils.js [hunk 2]
```diff
@@ -47,8 +47,5 @@ export function hasValidGlobalTemplatePersonalisation(personalisation) {
     return false;
   }
 
-  if (personalisationFields[0] !== "body") {
-    return false;
-  }
-  return true;
+  return personalisationFields[0] == "body";
```
Comment (aaron-downing) [inline sandbox/handlers/utils.js:50]:
```suggestion
  return personalisationFields[0] === "body";
```

Minor and complete edge case but this would return `true` if personalisationFields[0] equalled `true` when using `==`, love javascript

File: tests/lib/constants/constants.py [hunk 3]
```diff
@@ -42,8 +42,8 @@
 UNEXPECTED_429 = AssertionError('Unexpected 429')
 UNEXPECTED_504 = AssertionError('Unexpected 504')
 
-VALID_ACCEPT_HEADERS = ["*/*", DEFAULT_CONTENT_TYPE, "application/vnd.api+json"]
```
[the remark this answers is not in the document]
Comment (aaron-downing) [reply tests/lib/constants/constants.py:45]: Ohhh was `"application/vnd.api+json"` the contents of `DEFAULT_CONTENT_TYPE`?

File: sandbox/handlers/utils.js (hunk 2 above)
Comment (simonlabarere) [reply sandbox/handlers/utils.js:50]: Yep thanks for that, I come from java and `===` still confuses me a lot :D

File: tests/lib/constants/constants.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (simonlabarere) [reply tests/lib/constants/constants.py:45]: Correct, it was the same value twice.

**Review by aaron-downing — APPROVED** [review]

File: sandbox/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM node:12
+FROM node:20.15.0
```
Comment (Ian-Hodges) [inline sandbox/Dockerfile:1]: Any reason we can't go to 22?

Comment (Ian-Hodges) [reply sandbox/Dockerfile:1]: Ignore that. 22 is still in `current` mode.

**Review by Ian-Hodges — APPROVED** [review]

**Review by tdroza-nhs — APPROVED** [review]
